app/controllers/projects_controller.rb

```diff
@@ -56,8 +56,18 @@ def remove
   end
   
   def destroy
-    removed_project = Project.find_by_id(params[:admin][:id])
-    if Admin.find_by_id(session[:user_id]).try(:authenticate, params[:admin][:password])
+    if session[:user] == "admin"
+      removed_project = Project.find_by_id(params[:admin][:id])
+      check = Admin.find_by_id(session[:user_id]).try(:authenticate, params[:admin][:password])
+    elsif session[:user] == "instructor"
+      removed_project = Project.find_by_id(params[:instructor][:id])
+      check = Instructor.find_by_id(session[:user_id]).try(:authenticate, params[:instructor][:password])
+    else
+      flash[:warning] = "Unauthorized action"
+      redirect_to home_path
```

Comment:
Why are we redirecting to the home_path with a unauthorized action? Also note, instructor can't go to the home page without failing currently btw, something to note.